src/GitVersionTask/AssemblyInfoBuilder/AssemblyInfoBuilder.cs

```diff
@@ -32,26 +38,56 @@ static class GitVersionInformation
 {4}
 }}
 
-
-", 
-vars.AssemblySemVer,
- vars.MajorMinorPatch + ".0", 
- vars.InformationalVersion,
-            vars.CommitDate,
-            GenerateVariableMembers(vars));
+[System.Runtime.CompilerServices.CompilerGenerated]
+[AttributeUsage(AttributeTargets.Assembly)]
+sealed class GitVersionInformationAttribute : System.Attribute
+{{
+{5}
+}}",
+                                         vars.AssemblySemVer,
+                                         vars.MajorMinorPatch + ".0",
+                                         vars.InformationalVersion,
+                                         vars.CommitDate,
+                                         GenerateStaticVariableMembers(v),
+                                         GenerateAttributeVariableMembers(v))
+            .Trim();
 
         return assemblyInfo;
     }
 
-    string GenerateVariableMembers(IEnumerable<KeyValuePair<string, string>> vars)
+
+    static string GenerateStaticVariableMembers(IList<KeyValuePair<string, string>> vars)
     {
         var members = new StringBuilder();
```

Comment:
`return string.Join(Environment.NewLine, vars.Select(variable => string.Format("    public static string {0} = \"{1}\";", variable.Key, variable.Value)))`

One liner for this if you want